usr/local/www/system_gateways.php

```diff
@@ -100,8 +103,16 @@ function can_delete_gateway_item($id) {
 	if (is_array($config['staticroutes']['route'])) {
 		foreach ($config['staticroutes']['route'] as $route) {
 			if ($route['gateway'] == $a_gateways[$id]['name']) {
-				$input_errors[] = sprintf(gettext("Gateway '%s' cannot be deleted because it is in use on Static Route '%s'"), $a_gateways[$id]['name'], $route['network']);
-				break;
+				if ($disable) {
+					// The user wants to disable this gateway.
+					if (!isset($route['disabled'])) {
+						// But there is a static route that uses this gateway and is enabled (not disabled).
+						$input_errors[] = sprintf(gettext("Gateway '%s' cannot be disabled because it is in use on Static Route '%s'"), $a_gateways[$id]['name'], $route['network']);
+					}
+				} else {
+					// The user wants to delete this gateway, but there is a static route (enabled or disabled) that refers to the gateway.
+					$input_errors[] = sprintf(gettext("Gateway '%s' cannot be deleted because it is in use on Static Route '%s'"), $a_gateways[$id]['name'], $route['network']);
+				}
```

Comment:
There is also a test at line 93 using $disable - that reads OK to me. It would seem odd if the test here at line 106 was based on the reverse logic.
Should I reverse both tests? Or?